Packs/ArcherRSA/Integrations/ArcherV2/ArcherV2.py

```diff
@@ -220,19 +228,112 @@ def search_records_by_report_soap_request(token, report_guid):
+
+    Returns:
+        str: An XML string representing the FilterCondition element.
+    """
+    root = ET.Element(condition_type.value)
+
+    operator_element = ET.SubElement(root, 'Operator')
+    operator_element.text = operator
+
+    field_element = ET.SubElement(root, 'Field', attrib={'name': field_name})
+    field_element.text = field_id
+
+    value_element = ET.SubElement(root, 'Value')
+    value_element.text = search_value
+
+    for tag_name, tag_text in (sub_elements_tags_values or {}).items():
+        sub_element = ET.SubElement(root, tag_name)
+        sub_element.text = tag_text
+
+    return ET.tostring(root, encoding='unicode')
+
+
+def construct_content_filter_condition(operator: str, level_id: str, search_value: str) -> str:
+    """
+    Constructs an XML string representing a content filter condition for searching records.
+
+    Args:
+        operator (str): The comparison operator (e.g., 'Equals', 'GreaterThan').
+        field_name (str): The name of the application field.
+        field_id (str): The ID of the application field.
+        search_value (str): The value for the comparison.
+
+    Returns:
+        str: An XML string representing the ContentFilterCondition element.
+    """
+    root = ET.Element(FilterConditionTypes.content.value)
+
+    level = ET.SubElement(root, 'Level')
+    level.text = level_id
+
+    operator_element = ET.SubElement(root, 'Operator')
+    operator_element.text = operator
+
+    values_element = ET.SubElement(root, 'Values')
+    value_element = ET.SubElement(values_element, 'Value')
+    value_element.text = search_value
+
+    return ET.tostring(root, encoding='unicode')
+
+
+def construct_operator_logic(logical_operator: str | None, conditions_count: int) -> str:
+    """
+    Constructs an XML string representing the operator logic that applies to the filtering conditions for searching records.
+
+    Args:
+        logical_operator (str | None): The logical operator (e.g. 'AND', 'OR', 'XOR').
+        conditions_count (int): The number of conditions.
+
+    Returns:
+        str: An XML string representing the OperatorLogic element.
+    """
+    if not logical_operator or not logical_operator.strip():
+        return ''
+
+    # Only relevant if there are two or more conditions
+    if conditions_count < 2:
+        return ''
+
+    root = ET.Element('OperatorLogic')
+    root.text = f' {logical_operator.upper()} '.join([str(num) for num in range(1, conditions_count + 1)])
+    return ET.tostring(root, encoding='unicode')
+
+
 def search_records_soap_request(
-    token,
-    app_id,
-    display_fields,
-    field_id,
-    field_name,
-    search_value,
-    date_operator="",
-    field_to_search_by_id="",
-    numeric_operator="",
-    max_results=10,
-    level_id="",
+    token: str,
+    app_id: str,
+    display_fields: str,
+    field_id: str,
+    field_name: str,
+    search_value: str | None,
+    date_operator: str | None = "",
```

Comment:
Sure about `| None`? or is it always a string, possibly an empty one?

Reply:
Yes, `date_operator` is indeed of type `str | None` since it is used for two purposes:
* **`archer-search-records`** 
   * The command has an _optional_ `dateOperator` input argument since a user might not always want to filter by date
   * In this case, the variable is set as follows: `date_operator: str | None = args.get("dateOperator")`
* **`fetch-incidents`** 
   * Here we set the `date_operator` internally in the logic to a  _hardcoded string_ (`"GreaterThan"`) to fetch events after a certain timestamp
   * In this case, the variable is set as follows: `date_operator: str = "GreaterThan"`

I hope this clarifies things